WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift

```diff
@@ -615,13 +620,14 @@ extension OrderListViewController {
         return false
     }
 
-    func showOrderDetails(_ order: Order, onCompletion: ((Bool) -> Void)? = nil) {
+    func showOrderDetails(_ order: Order, shouldScrollIfNeeded: Bool = false, onCompletion: ((Bool) -> Void)? = nil) {
         let viewModel = OrderDetailsViewModel(order: order)
         switchDetailsHandler([viewModel], 0, true) { [weak self] hasBeenSelected in
             guard let self else { return }
             if hasBeenSelected {
-                onOrderSelected(id: order.orderID)
+                onOrderSelected(id: order.orderID, shouldScrollIfNeeded: shouldScrollIfNeeded)
             }
+            onCompletion?(hasBeenSelected)
```

Comment:
🤔 Interesting, so we were not calling the completion handler when `hasBeenSelected` returns false? 

Unrelated to this PR, but around the same logic: I find the current API for `switchDetailsHandler` very confusing/opaque, returning objects like `switchDetailsHandler([], 0, true, nil)` says nothing to the consumer of the API and perhaps we should improve the interface/signatures.

Reply:
> 🤔 Interesting, so we were not calling the completion handler when hasBeenSelected returns false?

Nope, we were _never_ calling it, not just when it was false.

> Unrelated to this PR, but around the same logic: I find the current API for switchDetailsHandler very confusing/opaque, returning objects like switchDetailsHandler([], 0, true, nil) says nothing to the consumer of the API and perhaps we should improve the interface/signatures. I'll add to my TODO list to investigate a bit around this and open an issue.

Agreed. It's from the original HACK week project so it would be good to improve it at some point, because it's not clear. A lot of the issues come from it being a closure, and not being able to name the parameters, but even if we simply wrapped some of the calls in friendlier function names, it would be an improvement.